File: challenges/2026-09-20-high-score-reel-combinations/solution.go
package main

import (
	"fmt"
	"reflect"
)

func highScoreCombinations(reels [][]byte, bannedPrefixes []string) []string {
	// TODO: implement
	return []string{}
}

type testCase struct {
	name     string
	reels    [][]byte
	banned   []string
	expected []string
}

func main() {
	tests := []testCase{
		{
			"example 1",
			[][]byte{{'A', 'B'}, {'C', 'D'}, {'E', 'F'}},
			[]string{"BD"},
			[]string{"ACE", "ACF", "ADE", "ADF", "BCE", "BCF"},
		},
		{
			"example 2",
			[][]byte{{'X'}, {'Y', 'Z'}, {'A', 'B', 'C'}},
			[]string{"XZ", "XYB"},
			[]string{"XYA", "XYC"},
		},
		{
			"example 3",
			[][]byte{{'A'}, {'B'}, {'C'}},
			[]string{},
			[]string{"ABC"},
		},
		{
			"edge: first letter banned prunes everything",
			[][]byte{{'A'}, {'B'}, {'C'}},
			[]string{"A"},
			[]string{},
		},
		{
			"edge: mid-level banned prefix prunes a branch",
			[][]byte{{'P', 'Q'}, {'R'}, {'S', 'T'}},
			[]string{"QR"},
			[]string{"PRS", "PRT"},
		},
	}

	for _, tc := range tests {
		got := highScoreCombinations(tc.reels, tc.banned)
		status := "FAIL"
		if reflect.DeepEqual(got, tc.expected) {
			status = "PASS"
		}
		fmt.Printf("[%s] %s: expected=%v actual=%v\n", status, tc.name, tc.expected, got)
	}
}

File: challenges/2026-09-20-high-score-reel-combinations/solution.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution
{
    public static List<string> HighScoreCombinations(List<List<char>> reels, List<string> bannedPrefixes)
    {
        // TODO: implement
        return new List<string>();
    }

    public static void Main(string[] args)
    {
        var tests = new List<(string Name, List<List<char>> Reels, List<string> Banned, List<string> Expected)>
        {
            ("example 1",
                new List<List<char>> { new() { 'A', 'B' }, new() { 'C', 'D' }, new() { 'E', 'F' } },
                new List<string> { "BD" },
                new List<string> { "ACE", "ACF", "ADE", "ADF", "BCE", "BCF" }),
            ("example 2",
                new List<List<char>> { new() { 'X' }, new() { 'Y', 'Z' }, new() { 'A', 'B', 'C' } },
                new List<string> { "XZ", "XYB" },
                new List<string> { "XYA", "XYC" }),
            ("example 3",
                new List<List<char>> { new() { 'A' }, new() { 'B' }, new() { 'C' } },
                new List<string>(),
                new List<string> { "ABC" }),
            ("edge: first letter banned prunes everything",
                new List<List<char>> { new() { 'A' }, new() { 'B' }, new() { 'C' } },
                new List<string> { "A" },
                new List<string>()),
            ("edge: mid-level banned prefix prunes a branch",
                new List<List<char>> { new() { 'P', 'Q' }, new() { 'R' }, new() { 'S', 'T' } },
                new List<string> { "QR" },
                new List<string> { "PRS", "PRT" }),
        };

        foreach (var (name, reels, banned, expected) in tests)
        {
            var actual = HighScoreCombinations(reels, banned);
            string status = actual.SequenceEqual(expected) ? "PASS" : "FAIL";
            Console.WriteLine($"[{status}] {name}: expected=[{string.Join(", ", expected)}] actual=[{string.Join(", ", actual)}]");
        }
    }
}
